src/api.cpp

```diff
@@ -4820,56 +4674,50 @@ cl_int CLVK_API_CALL clGetSupportedImageFormats(cl_context context,
 
     // Iterate over all known CL/VK format associations and report the CL
     // formats for which the Vulkan format is supported
-    for (auto const& clvkfmt : gFormatMaps) {
-        const cl_image_format& clfmt = clvkfmt.first;
-        auto const& fmt_support = clvkfmt.second;
-        if ((fmt_support.flags & flags) != flags) {
-            continue;
-        }
-        VkFormat format = fmt_support.vkfmt;
-        VkFormatProperties properties;
-        vkGetPhysicalDeviceFormatProperties(pdev, format, &properties);
-
-        cvk_debug("Vulkan format %d:", format);
-        cvk_debug("  linear : %s",
-                  vulkan_format_features_string(properties.linearTilingFeatures)
-                      .c_str());
-        cvk_debug("  optimal: %s", vulkan_format_features_string(
-                                       properties.optimalTilingFeatures)
-                                       .c_str());
-        cvk_debug(
-            "  buffer : %s",
-            vulkan_format_features_string(properties.bufferFeatures).c_str());
-
-        cvk_debug("Required format features %s",
-                  vulkan_format_features_string(required_format_feature_flags)
-                      .c_str());
-        VkFormatFeatureFlags features;
-        if (image_type == CL_MEM_OBJECT_IMAGE1D_BUFFER) {
-            features = properties.bufferFeatures;
-        } else {
-            // TODO support linear tiling
-            features = properties.optimalTilingFeatures;
-        }
-        if ((features & required_format_feature_flags) ==
-            required_format_feature_flags) {
+    const cl_channel_order channel_orders[] = {
+        CL_R,    CL_A,     CL_RG,    CL_RA,        CL_RGB,
+        CL_RGBA, CL_BGRA,  CL_ARGB,  CL_INTENSITY, CL_LUMINANCE,
+        CL_Rx,   CL_RGx,   CL_RGBx,  CL_DEPTH,     CL_DEPTH_STENCIL,
+        CL_sRGB, CL_sRGBx, CL_sRGBA, CL_sBGRA,     CL_ABGR,
+    };
+    const cl_channel_type channel_data_types[] = {
+        CL_SNORM_INT8,       CL_SNORM_INT16,        CL_UNORM_INT8,
+        CL_UNORM_INT16,      CL_UNORM_SHORT_565,    CL_UNORM_SHORT_555,
+        CL_UNORM_INT_101010, CL_SIGNED_INT8,        CL_SIGNED_INT16,
+        CL_SIGNED_INT32,     CL_UNSIGNED_INT8,      CL_UNSIGNED_INT16,
+        CL_UNSIGNED_INT32,   CL_HALF_FLOAT,         CL_FLOAT,
+        CL_UNORM_INT24,      CL_UNORM_INT_101010_2,
+    };
+    for (auto const image_channel_order : channel_orders) {
+        for (auto const image_channel_data_type : channel_data_types) {
```

Comment:
I'm not sure I understand why you changed the iteration logic. With the current tables, these loops will iterate over 340 combinations, regardless of the number of CL <-> Vulkan mappings. Adding a new format with a new channel order or data type now requires a change in two places. Is this just because the format map is no longer visible? If that's the case we could probably introduce a getter for it.

Reply:
I have introduced a getter for the map.